### PR TITLE
runtimevar: fix tests for blobvar and filevar on Windows

### LINKED ISSUE #1462: blob/fileblob: address TODO about windows
There's a TODO in the code:

```
// TODO: use filepath.FromSlash instead; and remove the pathSeparator arg
// from this function. Test Windows behavior by opening a bucket on Windows.
```

Address this.

It may also be in `filevar.go`.

### PR DESCRIPTION
**Pull request by vangent**
Updates #1306.
Fixes #1462.

### CODE REVIEW

File: runtimevar/filevar/filevar.go [hunk 1]
```diff
@@ -103,18 +103,11 @@ func (o *URLOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimeva
 	for param := range q {
 		return nil, fmt.Errorf("open bucket %q: invalid query parameter %q", u, param)
 	}
-	return New(mungeURLPath(u.Path, os.PathSeparator), o.Decoder, &o.Options)
-}
-
-func mungeURLPath(path string, pathSeparator byte) string {
-	if pathSeparator != '/' {
+	path := u.Path
+	if os.PathSeparator != '/' {
```
Comment (shantuo) [inline runtimevar/filevar/filevar.go]: I think you can use `filepath.Clean` to simplify this.

File: runtimevar/blobvar/blobvar_test.go [hunk 2]
```diff
@@ -105,6 +106,12 @@ func TestOpenVariable(t *testing.T) {
 	}
 	defer os.RemoveAll(dir)
 
+	// Convert dir to a URL path, adding a leading "/" if needed on Windows.
+	dirpath := filepath.ToSlash(dir)
+	if os.PathSeparator != '/' && !strings.HasPrefix(dirpath, "/") {
```
Comment (shantuo) [inline runtimevar/blobvar/blobvar_test.go]: Maybe add a comment saying that on Unix the path will be resolved to start with a '/'.

File: runtimevar/filevar/filevar.go (hunk 1 above)
Comment (vangent) [reply runtimevar/filevar/filevar.go]: I don't think that will strip the "/" prefix. It also does a bunch of other things I'm not sure I want.

File: runtimevar/blobvar/blobvar_test.go (hunk 2 above)
Comment (vangent) [reply runtimevar/blobvar/blobvar_test.go]: Done.

**Pull request merged**